### DISCUSSION TITLE
Add HS code in PrestaShop (classifying traded products)  - Harmonized System Codes

### DISCUSSION THREAD

**Discussion by karsttuinstra** [PrestaShop/PrestaShop#40182] [Ideas]
#### Is your feature request related to a problem?

As you can read in a blog post from Prestashop themselves the HS code is very important for international shipments and this importance will only increase due to ITMATT, Brexit and the VAT abolishment:
https://www.prestashop.com/en/blog/importance-transparency-cross-border-transactions

#### Describe the solution you'd like

Add field for HS code under Products - Shipping

#### Alternatives you've considered

We now have to build something hacky in our carrier module.

[a comment by an automated account is not shown]

**Comment by florine2623** [comment]
Hi @karsttuinstra ,
Thansk for your request.
I will ping our Product Team, they will make a decision regarding your request.
Thanks !

**Comment by florine2623** [comment]
Hi @PrestaShop/prestashop-product-team

@karsttuinstra would like to have added fields for Harmonized System Code. The article posted in 2015 on our blog suggests the use of those for international trade. 

What do you think about this request ?

Thanks !

**Comment by JoryHogeveen** [comment]
Any update on this?

**Comment by matks** [comment]
Hi, looks like Product Team did not have the time to look at this issue yet.

However I have 2 questions:
- I dont see any information about HS code in https://www.prestashop.com/en/blog/choosing-a-online-payment-solution are you sure this is the right blog article
- I'm a developer, not a PM ; so I dont what HS code is. I google it, so if I get it right it's a number assigned to a product ; and it tells you what VAT must be applied, depending on the country?

**Comment by JoryHogeveen** [comment]
Hi @matks 

More info about HS codes and EORI numbers here: https://www.prestashop.com/en/blog/brexit-vat

So, alongside the VAT module for the Brexit, these type of info is also important regarding the Brexit. Magento and other systems are already implementing these type of codes so I believe Prestashop shouldn't be lagging behind.

**Comment by NKoonen** [comment]
Since Prestashop is really behind on this issue: https://www.prestashop.com/forums/topic/1046735-free-modulehs-codes

**Comment by BenTen** [comment]
> 
> 
> Since Prestashop is really behind on this issue: https://www.prestashop.com/forums/topic/1046735-free-modulehs-codes

@NKoonen Why don't you propose your work here? Can be a good thing to add natively in PS.

**Comment by copostic** [comment]
Nice idea but wouldn't it be better by combination ?
For example I have a product that is in silver or gold depending what combination customer choose so HS.Code is not the same ..

**Comment by JoryHogeveen** [comment]
As far as I know HS codes are more product "category" related so variations should have the same HS code right?

**Comment by BenTen** [comment]
I guess it's just an example:
For sure color will be the same HS Code, but a shop can propose a box with 2 combinations.
The first in metal and the second in wood.

The first has the HS Code 73101010 and the second 44151000

**Comment by tswfi** [comment]
We would also need this for few carrier modules. Almost all carriers are starting to require this information for shipments.

**Comment by JoryHogeveen** [comment]
@BenTen Ah right, good example!

**Comment by 0dp** [comment]
Besides HS Code we are required to inform the goods Country of Origin.

**Comment by NKoonen** [comment]
> Besides HS Code we are required to inform the goods Country of Origin.

This function has been added to my module [here](https://github.com/NKoonen/hscodes).

**Comment by dangelion** [comment]
Are they implementing this in Prestashop by default?

**Comment by seederp2p** [comment]
This is very important.

Please add it to prestashop.

Another blog post from Prestashop themselves where they state the importance of HS codes.

https://prestashop.com/blog/resources/the-challenges-of-cross-border-shipping-tips-for-a-smooth-journey/

**Comment by Betafer** [comment]
I hope to see the module integrated into prestashop as a native module!

**Comment by Spectrom66** [comment]
finally a module available!
https://addons.prestashop.com/en/accounting-invoicing/96104-hs-codes-international-customs-nomenclature.html

**Comment by JoryHogeveen** [comment]
@Spectrom66 You find a free one here: https://github.com/NKoonen/hscodes

**Comment by pppplus** [comment]
@NKoonen 
Do you plane to update your module, to be compatible with PS 9.0 ?
